src/vm/threads.h

```diff
@@ -4900,7 +4900,7 @@ class Thread: public IUnknown
     bool GetGCModeOnSuspension()
     {
         LIMITED_METHOD_CONTRACT;
-        return m_gcModeOnSuspension != 0;
+        return m_gcModeOnSuspension != (ULONG)0;
```

Comment:
This is really unfortunate behavior of GCC, I think it is a bug in the template handling. I've verified that if the `m_gcModeOnSuspension` was ULONG, it would not require any cast. If the argument of `Volatile<T>::operator!=` was not const, it would also not require any cast. 
Could you please make it less ugly by using `0U` instead of the cast?

But I also wonder how many cases like this we have in our code base. If there are many, it seems it would make sense to try to somehow change the operator definition for GCC so that we don't have to fix all the usages.

Reply:
I'm able to do %99 compilation without any warnings by suppressing pragma warnings at this moment and MIDL/jit stuff stubbed.

https://github.com/franksinankaya/coreclr/tree/gcc_compile_rebase